support-frontend/assets/helpers/abTests/abtest.ts

```diff
@@ -224,16 +224,39 @@ function getAmountsTestParticipations(
 		return urlTest;
 	}
 
-	const { test } = settings.amounts?.[countryGroupId] ?? {};
+	if (!settings.amounts) {
+		return null;
+	}
+
+	const AMOUNTS = settings.amounts;
+
+	// Two-tier amounts - check for country test, then region test
+	let testArray = AMOUNTS.filter((t) => t.target === country);
+	// If there is a country level test we need to discount it if its isLive boolean is false
+	if (!testArray.length || !testArray[0].isLive) {
+		testArray = AMOUNTS.filter((t) => t.target === countryGroupId);
+	}
```

Comment:
Similar to the SDC PR, I wonder if we could use `find` here and lose the mutation:

```js
const matchedTest = AMOUNTS.find((t) => t.target === country && t.isLive) || AMOUNTS.find((t) => t.target === countryGroupId);
```

You could even create named functions for the two finds, which I think makes the country > region precedence nice and clear:

```js
const findMatchingCountryTest = (amounts, country) => amounts.find(t => t.target === country && t.isLive);

const findMatchingRegionTest = (amounts, countryGroupId) => amounts.find(t => t.target === countryGroupId);

const matchedTest = findMatchingCountryTest(amounts, country) || findMatchingRegionText(amounts, countryGroupId);
```

Reply:
I'll experiment for clarity

Reply:
I've updated with the same `find` code as used in SDC (not the `findMatching functions` suggestion).